src/main/java/io/reactivex/internal/operators/flowable/FlowablePublish.java

```diff
@@ -557,12 +557,20 @@ void dispatch() {
                         InnerSubscriber<T>[] freshArray = subscribers.get();
                         if (subscribersChanged || freshArray != ps) {
                             ps = freshArray;
+
+                            // if we did emit at least one element, request more to replenish the queue
+                            if (d != 0) {
+                                if (sourceMode != QueueSubscription.SYNC) {
+                                    upstream.get().request(d);
+                                }
+                            }
+
                             continue outer;
                         }
                     }
 
                     // if we did emit at least one element, request more to replenish the queue
-                    if (d > 0) {
+                    if (d != 0) {
```

Comment:
Sorry I don't understand this particular change, how can we end up with negative `d`?

Otherwise it's same as previously used condition 🤔

Reply:
`d` should be never negative, thus `!=` is just fine.

Reply:
but then `d > 0` is okay too, or is this a performance optimization?

Reply:
It is slighly faster.